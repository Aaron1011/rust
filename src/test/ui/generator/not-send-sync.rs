#![feature(generators)]

use std::cell::Cell;

fn main() {
    fn assert_sync<T: Sync>(_: T) {}
    fn assert_send<T: Send>(_: T) {}

    assert_sync(|| {
        let a = Cell::new(2);
        yield;
    });

    let a = Cell::new(2);
    assert_send(|| {
        //~^ ERROR: E0277
        drop(&a);
        yield;
    });
}
