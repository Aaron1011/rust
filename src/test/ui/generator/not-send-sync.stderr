error[E0277]: `std::cell::Cell<i32>` cannot be shared between threads safely
  --> $DIR/not-send-sync.rs:15:5
   |
LL |     fn assert_send<T: Send>(_: T) {}
   |        -----------    ---- required by this bound in `main::assert_send`
...
LL |     assert_send(|| {
   |     ^^^^^^^^^^^ `std::cell::Cell<i32>` cannot be shared between threads safely
   |
   = help: the trait `std::marker::Sync` is not implemented for `std::cell::Cell<i32>`
   = note: required because of the requirements on the impl of `std::marker::Send` for `&std::cell::Cell<i32>`
   = note: required because it appears within the type `[generator@$DIR/not-send-sync.rs:15:17: 19:6 a:&std::cell::Cell<i32> _]`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
